docs/guide/api-environment-runtimes.md

```diff
@@ -363,29 +363,24 @@ export const runner = new ModuleRunner(
 await runner.import('/entry.js')
 ```
 
-In this case, the hot channel only needs to implement the `setInvokeHandler`:
+In this case, the `handleInvoke` method in the `NormalizedHotChannel` can be used:
 
 ```ts
-let invokeHandler: HotChannelInvokeHandler | undefined
-
-const httpHotChannel = {
-  setInvokeHandler(_invokeHandler) {
-    invokeHandler = _invokeHandler
-  },
-}
-// pass httpHotChannel to environment
-
-server.onRequest((request: Request) => {
-  const url = new URL(request.url)
-  if (url.pathname === '/invoke') {
-    if (invokeHandler) {
-      const payload = (await request.json()) as HotPayload
-      const result = invokeHandler?.(payload)
-      return new Response(JSON.stringify(result))
-    }
+class CustomDevEnvironment extends DevEnvironment {
```

Comment:
Not sure if it's a good idea to encourage extending environments. Maybe turn the example into something like this?

```ts
const environment = vite.environments.custom
server.onRequest(() => ... )
```